semantics/src/main/kotlin/com/strumenta/kolasu/semantics/symbol/description/SymbolDescription.kt

```diff
@@ -0,0 +1,46 @@
+package com.strumenta.kolasu.semantics.symbol.description
+
+import com.strumenta.kolasu.language.KolasuLanguage
+import com.strumenta.kolasu.model.Named
+import com.strumenta.kolasu.model.Node
+
+data class SymbolDescription(
+    override val name: String,
```

Comment:
What happens if I specify a field called "name"? Would this overlap with the name field or be separated?

Reply:
The two fields would be separated - we could handle the alignment in the DeclarativeSymbolProvider API. For example, whenever including a property named `name`, we could also se the name of the symbol. At the moment, the name can be set as follows:
```kotlin
object MySymbolProvider: DeclarativeSymbolProvider(
    symbolFor(SomeNode::class) {
        name("some name") // <- this sets the name of the symbol description
        include(SomeNode::name) // <- we could also implicitly set the name when including properties like these
    }
)
```
The `name()` directive could be removed, but the symbol description name would be null if no `name` property is included. What would happen with cases where a name property is not defined but we still want to define a symbol name. Perhaps we can keep both and specify that the last writer wins in the documentation. In the example above, the final name value will be given by `SomeNode::name`. What do you think?